girder/api/v1/assetstore.py

```diff
@@ -248,3 +250,73 @@ def getAssetstoreFiles(self, assetstore, limit, offset, sort, params):
         return list(self.model('file').find(
             query={'assetstoreId': assetstore['_id']},
             offset=offset, limit=limit, sort=sort))
+
+    @access.admin
+    @autoDescribeRoute(
+        Description('get space usage statistics by collection and by user.')
+        .modelParam('id', model='assetstore')
+        .errorResponse()
+        .errorResponse('You are not an administrator.', 403)
+    )
+    def getSpaceUsageStatistics(self, assetstore, params):
+        filesInStore = list(self.model('file').find(
+            query={'assetstoreId': assetstore['_id']}))
+
+        itemsUsingStore = list(self.model('item').find(query={
+            '_id': {
+                '$in': [x['itemId'] for x in filesInStore]
```

Comment:
This will need to change also, so that the size of the query sent to mongo is not O(n) with the number of files in the assetstore.